src/unix/unix.c

```diff
@@ -232,18 +232,25 @@ define_closure_function(1, 1, context, unix_fault_handler,
     } else if (is_page_fault(ctx->frame)) {
         pf_debug("page fault, vaddr 0x%lx, ctx %p, type %d, pc 0x%lx",
                  vaddr, ctx, ctx->type, ctx->frame[SYSCALL_FRAME_PC]);
-        vmap vm = vmap_from_vaddr(t->p, vaddr);
+        vmap vm;
+        if (vaddr >= PAGESIZE && vaddr < USER_LIMIT)
+            vm = vmap_from_vaddr(t->p, vaddr);
+        else
+            vm = INVALID_ADDRESS;
```

Comment:
It seems to me we could do without the (vaddr >= PAGESIZE && vaddr < USER_LIMIT) check and just leave `vmap vm = vmap_from_vaddr(t->p, vaddr)` as it was before: if vaddr is not between PAGESIZE and USER_LIMIT, vmap_from_vaddr() will return INVALID_ADDRESS.
Besides, there is a `if (vaddr >= USER_LIMIT) goto bug` at the beginning of this function (which in my opinion should not be there), so vaddr cannot be < USER_LIMIT here.

Reply:
I realize the vmap lookup should fail for out of bounds addresses, but I think the negligible cost of this range check is worth the protection against the risk that t->p->vmaps could ever contain bad entries, however minor, as such a condition might possibly create a path to an exploit. I removed the vaddr check at the top of the function, though.